plugins/woocommerce/client/legacy/js/frontend/geolocation.js

```diff
@@ -83,17 +83,14 @@ jQuery( function( $ ) {
 		// Updates our (cookie-based) cache of the hash value. Expires in 1 hour.
 		Cookies.set( 'woocommerce_geo_hash', hash, { expires: 1 / 24 } );
 
-		var this_page = window.location.toString();
-
-		if ( this_page.indexOf( '?v=' ) > 0 || this_page.indexOf( '&v=' ) > 0 ) {
-			this_page = this_page.replace( /v=[^&]+/, 'v=' + hash );
-		} else if ( this_page.indexOf( '?' ) > 0 ) {
-			this_page = this_page + '&v=' + hash;
-		} else {
-			this_page = this_page + '?v=' + hash;
-		}
+		const urlQuery     = new URL( window.location ).searchParams;
+		const existingHash = urlQuery.get( 'v' )
```

Comment:
Should this have a `;`?

Reply:
Good catch! Added in 1d6f912b2524960b3b7d4cffe66920861463d2b1.